client/python/cryoet_data_portal/src/cryoet_data_portal/_file_tools.py

```diff
@@ -54,15 +54,27 @@ def download_https(
 
 def get_destination_path(
     url: str,
-    dest_path: Optional[str],
+    dest_path: Optional[str] = None,
     recursive_from_prefix: Optional[str] = None,
 ) -> str:
+    """
+    Get the destination path for a file download.
+
+    Args:
+        url (str): The URL to download
+        dest_path (str): The destination path the files will download to
+        recursive_from_prefix (str): The prefix to use for recursive downloads
+
+    Returns:
+        str: The destination path for the file download
+
+    """
     if not dest_path:
         dest_path = os.getcwd()
     dest_path = os.path.abspath(dest_path)
 
-    if not os.path.isdir(dest_path) and recursive_from_prefix:
-        raise ValueError("Recursive downloads require a base directory")
+    if not os.path.isdir(dest_path):
+        os.makedirs(dest_path)
```

Comment:
When the dest_path is None, it should point to the working directory, and will pass this check. So, the only time this exception would show up would be when the user explicitly provides an invalid value and we should error on it. I think the error message could be more verbose in explaining the issue.  cc: @kandarpksk 

Also, [this line](https://github.com/chanzuckerberg/cryoet-data-portal/pull/940/files#diff-63d5c8af96144c2f98bba0f8110bc718612219f91e9b97e9b40e3d74a9a9bf73R85) should already be creating the recursive directory structure.

Reply:
So error when an invalid name is provided? Invalid meaning illegal path characters or maybe a path we can't write to?